pixeltable/catalog/table.py

```diff
@@ -77,6 +77,16 @@ def __init__(self, id: UUID, dir_id: UUID, name: str, tbl_version_path: TableVer
         self._tbl_version = None
 
     def _move(self, new_name: str, new_dir_id: UUID) -> None:
+        old_name = self._name
+        old_dir_id = self._dir_id
+
+        cat = catalog.Catalog.get()
+
+        @cat.register_undo_action
+        def _() -> None:
```

Comment:
Leave a todo to deal with this in a more complete manner. move() should fail if the table got renamed concurrently, at which point we need to reload the Table instance. We shouldn't try to fix up Table instances, we need to invalidate them and force a reload.